### ISSUE TITLE
Custom fields for departments

### ISSUE THREAD

**Issue by jxckUK** [SnailyCAD/snaily-cadv4#1281]
### Describe the feature

It would be extremely helpful if we could add custom fields to departments in order to store data such as:

- A colour code for in game spawn menus and blips
- Number values for internal scripting usage
- Table values such as a table of station coordinates in which we can use to spawn players at

As these fields are likely to be varied per community, instead of requesting these fields to be added individually can we get broad access to add custom fields to our own requirements such as the ones above.

**Comment by Dev-CasperTheGhost** [comment]
I can add a simple column `extraValues` with `Json` value?

**Comment by jxckUK** [comment]
> I can add a simple column `extraValues` with `Json` value?

I mean that would work fundamentally but I'd have issues writing partial data to a json value without first storing the original json value and *adding/amending* values within? Could add extra complexity my side but if that's the easiest/best solution on the CAD side I can work with it.

**Comment by Dev-CasperTheGhost** [comment]
cc @jxckUK

**Comment by Dev-CasperTheGhost** [comment]
Right, but I meant the JSON structure of that 😅

**Comment by Dev-CasperTheGhost** [comment]
> Ah, for all of them

Yeah, I'm not going to be making a field for each of these types. I'm going to make 1 field where you can store this extra information.

**Comment by jxckUK** [comment]
> > Ah, for all of them
> 
> Yeah, I'm not going to be making a field for each of these types. I'm going to make 1 field where you can store all the extra information.

Oh I see, do we need to expand on all the extra information or can we make it so you can store any additional information in this field?

**Comment by Dev-CasperTheGhost** [comment]
it will function like the appearance field actually
